### PR TITLE
Php 8

### PR DESCRIPTION
**Pull request by rodrigoaguilera**
### Disposition
This pull request:

- [x] Adds a feature
- [x] Has tests that cover changes

### Summary
Add php 8 to travis and changes second argument of call_user_func_array() from associative to indexed array. 
### Description
Before PHP 8 both types of arrays work the same but on PHP 8 associative arrays are interpreted as "named parameters". 
I am trying to fix what I find in my way of running robo 3 with php 8.

### CODE REVIEW

**Comment by hussainweb** [comment]
I came here to create a PR for this same bug. This is the error I see:
```
 [error]  Error: Unknown named parameter $options in call_user_func_array() (line 257 of /var/www/html/vendor/consolidation/annotated-command/src/CommandProcessor.php) #0 /var/www/html/vendor/consolidation/annotated-command/src/CommandProcessor.php(257): call_user_func_array(Array, Array)
#1 /var/www/html/vendor/consolidation/annotated-command/src/CommandProcessor.php(212): Consolidation\AnnotatedCommand\CommandProcessor->runCommandCallback(Array, Object(Consolidation\AnnotatedCommand\CommandData))
#2 /var/www/html/vendor/consolidation/annotated-command/src/CommandProcessor.php(176): Consolidation\AnnotatedCommand\CommandProcessor->validateRunAndAlter(Array, Array, Object(Consolidation\AnnotatedCommand\CommandData))
#3 /var/www/html/vendor/consolidation/annotated-command/src/AnnotatedCommand.php(311): Consolidation\AnnotatedCommand\CommandProcessor->process(Object(Symfony\Component\Console\Output\ConsoleOutput), Array, Array, Object(Consolidation\AnnotatedCommand\CommandData))
```

The fix with `array_values` works well.

**Review by hussainweb — APPROVED** [review]

**Comment by greg-1-anderson** [comment]
It doesn't look like the `nightly` test built. I'm +1 on the change to CommandProcessor.php, but I'd rather not add something to the Travis config unless it is actually fuctional.

**Comment by rodrigoaguilera** [comment]
I just removed the `nightly` key from travis. I don't know much about travis but maybe is only running the config from the main branch so not any Pull requester can change the config.
Better let the maintainers decide when to introduce php8 testing.

**Comment by greg-1-anderson** [comment]
I'd love to include php8 testing on every PR. Not going to hold up this PR for that, though. Thanks for the adjustment.

**Pull request merged**
